src/BenchmarkDotNet/Toolchains/MonoAotLLVM/MonoAotLLVMGenerator.cs

```diff
@@ -0,0 +1,88 @@
+
+namespace BenchmarkDotNet
+{
+    public class MonoAotLLVMGenerator : CsProjGenerator
+    {
+        private readonly string CustomRuntimePack;
+        private readonly string AotCompilerPath;
+        private readonly string AotCompilerMode;
+
+        public MonoAotLLVMGenerator(string targetFrameworkMoniker, string cliPath, string packagesPath, string customRuntimePack, string aotCompilerPath, string aotCompilerMode)
+            : base(targetFrameworkMoniker, cliPath, packagesPath, runtimeFrameworkVersion: null)
+        {
+            CustomRuntimePack = customRuntimePack;
+            AotCompilerPath = aotCompilerPath;
+            AotCompilerMode = aotCompilerMode;
+        }
+
+        protected override void GenerateProject(BuildPartition buildPartition, ArtifactsPaths artifactsPaths, ILogger logger)
+        {
+            BenchmarkCase benchmark = buildPartition.RepresentativeBenchmarkCase;
+            var projectFile = GetProjectFilePath(benchmark.Descriptor.Type, logger);
+
+            string useLLVM = "false";
+            if (AotCompilerMode == "llvm")
+            {
+                useLLVM = "true";
+            }
+
+            using (var file = new StreamReader(File.OpenRead(projectFile.FullName)))
+            {
+                var (customProperties, sdkName) = GetSettingsThatNeedsToBeCopied(file, projectFile);
+
+                string content = new StringBuilder(ResourceHelper.LoadTemplate("MonoAOTLLVMCsProj.txt"))
+                    .Replace("$PLATFORM$", buildPartition.Platform.ToConfig())
+                    .Replace("$CODEFILENAME$", Path.GetFileName(artifactsPaths.ProgramCodePath))
+                    .Replace("$CSPROJPATH$", projectFile.FullName)
+                    .Replace("$TFM$", TargetFrameworkMoniker)
+                    .Replace("$PROGRAMNAME$", artifactsPaths.ProgramName)
+                    .Replace("$COPIEDSETTINGS$", customProperties)
+                    .Replace("$CONFIGURATIONNAME$", buildPartition.BuildConfiguration)
+                    .Replace("$SDKNAME$", sdkName)
+                    .Replace("$RUNTIMEPACK$", CustomRuntimePack ?? "")
+                    .Replace("$COMPILERBINARYPATH$", AotCompilerPath)
+                    .Replace("$RUNTIMEIDENTIFIER$", buildRuntimeIdentifier())
+                    .Replace("$USELLVM$", useLLVM)
+                    .ToString();
+
+                File.WriteAllText(artifactsPaths.ProjectFilePath, content);
+            }
+        }
+
+        private string buildRuntimeIdentifier()
+        {
+            string os = "";
+
+            if (RuntimeInformation.IsLinux())
+            {
+                os = "linux";
+            }
+            else if (RuntimeInformation.IsMacOSX())
+            {
+                os = "osx";
+            }
+            else if (RuntimeInformation.IsWindows())
+            {
+                os = "win";
+            }
+
+            return $"{os}-{RuntimeEnvironment.RuntimeArchitecture}";
+        }
```

Comment:
we already have such a method, please reuse it: 

https://github.com/dotnet/BenchmarkDotNet/blob/502dc9aca86f21a9d38da9634f3bec2143f1f6f5/src/BenchmarkDotNet/Toolchains/DotNetCli/CustomDotNetCliToolchainBuilder.cs#L136-L144

Reply:
Since that method is protected, what is the best way to reuse it? Just make it public? It doesn't seem like I need to use the whole tool chain builder.

Reply:
You can make it `internal`